### ISSUE TITLE
Ease running unit tests with packaging "eclipse-plugin-test"

### ISSUE THREAD

**Issue by kwin** [eclipse-tycho/tycho#1491]
Currently there is no unit test execution bound by default for packaging`eclipse-plugin-test` (https://github.com/eclipse-tycho/tycho/blob/master/tycho-maven-plugin/src/main/resources/META-INF/plexus/components.xml) but some tests should not be executed inside Eclipse as IT but rather  as plain unit test only on the Maven side with regular "maven-surefire-plugin". Although this is supported already for `eclipse-plugin` (https://github.com/eclipse-tycho/tycho/commit/7d8d4b8cf826dbac55dc2696e61de46a5c2cefa5) it is not yet for `eclipse-plugin-test`.
Currently it is some effort to those up due to the missing default binding (https://stackoverflow.com/a/44926613).
Please provide some default binding for unit tests and make sure that those don't conflict with the PDE execution, i.e. they don't end up in the packaged bundle. This requires some custom configuration to place them somewhere outside the regular `src` folder, e.g. in `test` instead of the regular `src/main/test`). That should be possible by adjusting the `${project.build.testSourceDirectory}` (https://maven.apache.org/pom.html#directories) and test resources (https://maven.apache.org/pom.html#Resources).

**Comment by laeubi** [comment]
This is already supported pleas see:
- https://github.com/eclipse-tycho/tycho/blob/master/RELEASE_NOTES.md#support-resolving-of-junit-classpath-container
- https://github.com/eclipse-tycho/tycho/blob/master/RELEASE_NOTES.md#execute-unit-tests-with-eclipse-plugin-packaging

**Comment by kwin** [comment]
> So eclipse-plugin-test is not necessary at all.

This issue is particularly about supporting `eclipse-plugin-test`. For certain edge cases it is useful to have these contain unit tests as well (e.g. unit testing code that is used inside the ITs). I don't think that it does do harm to add the lifecycle binding also to `eclipse-plugin-test`.

**Comment by laeubi** [comment]
You can do with `eclipse-plugin` do anything you can do with `eclipse-plugin-test`, if you like you can even configure the mojos used in  `eclipse-plugin` for `eclipse-plugin-test` but adding any new behavior to `eclipse-plugin-test` is not desired as it already complicates things in its current state.

**Comment by kwin** [comment]
> but adding any new behavior to eclipse-plugin-test is not desired

I don't really have the feeling that this is a community driven project. Usually in open source one expresses individual opinions, but in the case of tycho and m2e it feels that @laeubi just decides what is useful and what isn't without giving other committers the chance to comment. In general closing feature requests within hours/days with not at least one other  committer expressing the same opinion feels very rude and doesn't particularly encourage people to contribute here.

**Comment by laeubi** [comment]
> I don't really have the feeling that this is a community driven project. Usually in open source one expresses individual opinions, but in the case of tycho and m2e it feels that @laeubi just decides what is useful and what isn't without giving other committers the chance to comment.

Any committer can (and will) give independent opinions and even can disagree and then we can discuss and came to other conclusions, but there are is no need for every committer to express individual agreement on every particular issue, so if other agree there is no need to do so. There are often other things where I do not participate because I'm just fine with that.

Also open source do not mean that each an every request must considered, and as there are already functions covering the use-case it is fine than keeping things open for ever.

So if you need the same behavior "for special circumstances" you can already do that (even though its quite pointless) but you should not assume that this is useful for all users of the project.

Especially for this particular example, my personal intend was to completely **remove** `eclipse-plugin-test` as it is nowadays more or less obsolete (beside some technical variants that might be solvable), but other committers disagreed and find that the current behavior of `eclipse-plugin-test` should be retained and I'm fine with that even though I do not agree.

Nerveless it makes no sense to **duplicate** the same functionality that is offered by `eclipse-plugin` so if you require *that* behavior, simply use `eclipse-plugin` as there is nothing `eclipse-plugin-test` can do better here, and in that regard you can maybe trust the person that has implemented that particular feature for Tycho even though most other committer assumed **it is not useful at all** ...

**Comment by kwin** [comment]
Closing an issue IMHO ends a discussion so I would recommend to at least wait 2 weeks before closing issues.

**Comment by laeubi** [comment]
@kwin if you like to _discuss_ something its generally better to open a new https://github.com/eclipse-tycho/tycho/discussions (issues are used for bugs or actually implement a feature), and if there is a decision one can easily create an issue from a discussion.

So if you like I can convert the issue into a discussion, or you simply open a new one with maybe rephrased question/text, e.g. it should at best contain _why_ you suggest a change (e.g. what can be done with `eclipse-plugin-test` what can not already be archived with `eclipse-plugin`) in contrast of _how_ this would be implemented (maybe there are other options that do not require a change or we could solve the user-story better).
